pkg/smokescreen/smokescreen.go

```diff
@@ -40,7 +40,9 @@ type ctxUserData struct {
 	decision *aclDecision
 }
 
-type denyError error
+type denyError struct {
```

Comment:
Why is this form preferred over the other?

Reply:
We're defining `denyError` because we want to later check if a particular `error` is a `denyError`.  But the first one just creates an alias, and because `error` is an interface, *any* `error` is also a `denyError` -- type assertions for interfaces are just duck-typing.

Here's a simplified example: https://play.golang.org/p/X7hny-SZ6a0